src/plugins/vtt-thumbnails/index.js

```diff
@@ -89,179 +85,65 @@ const VttThumbnailsPlugin = (function () {
+      });
   };
 
   VttThumbnailsPlugin.prototype.setupThumbnailElement = function setupThumbnailElement() {
-    let _this4 = this;
-
-    let mouseDisplay = this.player.$('.vjs-mouse-display');
-    this.progressBar = this.player.$('.vjs-progress-control');
-    let thumbHolder = this.player.$('.vjs-vtt-thumbnail-display') || document.createElement('div');
-    thumbHolder.innerHTML = '';
-    thumbHolder.setAttribute('class', 'vjs-vtt-thumbnail-display');
-    this.progressBar.appendChild(thumbHolder);
-    this.thumbnailHolder = thumbHolder;
-    let timeHolder = document.createElement('div');
-    timeHolder.setAttribute('class', 'vjs-vtt-time-display');
-    this.thumbnailHolder.appendChild(timeHolder);
-    this.timeHolder = timeHolder;
-    if (mouseDisplay) {
-      mouseDisplay.classList.add('vjs-hidden');
-    }
-    this.registeredEvents.progressBarMouseEnter = function () {
-      return _this4.onBarMouseenter();
-    };
-    this.registeredEvents.progressBarMouseLeave = function () {
-      return _this4.onBarMouseleave();
-    };
-    this.progressBar.addEventListener(
-      'mouseenter',
-      this.registeredEvents.progressBarMouseEnter
-    );
-    this.progressBar.addEventListener(
-      'mouseleave',
-      this.registeredEvents.progressBarMouseLeave
-    );
-  };
-
-  VttThumbnailsPlugin.prototype.onBarMouseenter = function onBarMouseenter() {
-    let _this5 = this;
-
-    if (this.progressBar.offsetWidth < 400) {
-      // Don't show in small players
-      return;
-    }
-
-    this.mouseMoveCallback = function (e) {
-      _this5.onBarMousemove(e);
+    const mouseDisplay = this.player.$('.vjs-progress-holder .vjs-mouse-display');
+    this.thumbnailHolder = this.player.$('.vjs-vtt-thumbnail-display') || videojs.dom.createEl('div', undefined, {
+      class: 'vjs-vtt-thumbnail-display',
+      style: `width: ${this.vttData[0].css.width}`
+    });
+    this.thumbnailHolder.innerHTML = '';
+    this.thumbnailHolder.setAttribute('class', 'vjs-vtt-thumbnail-display');
+    mouseDisplay.appendChild(this.thumbnailHolder);
+
+    const halfThumbnailWidth = parseInt(this.vttData[0].css.width, 10) / 2;
+
+    const updateThumbnailStyle = this.updateThumbnailStyle.bind(this);
+
+    this.player.getDescendant([
+      'controlBar',
+      'progressControl',
+      'seekBar',
+      'mouseTimeDisplay'
+    ]).update = function (seekBarRect, seekBarPoint) {
+      const time = seekBarPoint * this.player_.duration();
+      const width = seekBarRect.width;
+
+      updateThumbnailStyle(time);
+
+      this.getChild('timeTooltip').updateTime(seekBarRect, seekBarPoint, time, () => {
+        if ((seekBarRect.width * seekBarPoint) < halfThumbnailWidth) {
+          this.el_.style.left = halfThumbnailWidth;
+        } else if ((seekBarRect.width * seekBarPoint) + halfThumbnailWidth > width) {
+          this.el_.style.left = (seekBarRect.width * seekBarPoint) - halfThumbnailWidth;
+        } else {
```

Comment:
it could be because i am not familiar with the codebase, but i am having a hard time understand why this specific positioning, for that reason i would add comments about why this is needed

Reply:
we want it to stop "moving" before the thumbnails exits the player which is at `halfThumbnailWidth` distance from the edge.

Reply:
Added a comment